### ISSUE TITLE
Custom Search-Index implementation cannot set the name

### ISSUE THREAD

**Issue by HellPat** [statamic/cms#1200]
I'm currently writing my own search implementation.
When I use a custom creator which is called via `->callCustomCreator`  I have no access to the name which is getting resolved.

https://github.com/statamic/cms/blob/v3.0.0-beta.9/src/Support/Manager.php#L31-L44

For the internal drivers `Algolia` and `Comb` the name is beeing passed as second argument.
(only a few lines later)
https://github.com/statamic/cms/blob/v3.0.0-beta.9/src/Support/Manager.php#L49

I registered the Index like that:

```php
    public function register()
    {
        /** @var IndexManager $indexManager */
        $indexManager = $this->app->get(IndexManager::class);

        $indexManager->extend('database', function ($app, $config) {
            return new class('here_should_be_the_name_but_i_have_no_access_to_it_because_$config_provides_the_values_only_but_not_the_key', $config) extends Index {
                public function search($query)
                {
                    // TODO: Implement search() method.
                }

                protected function insertDocuments(Documents $documents)
                {
                    // TODO: Implement insertDocuments() method.
                }

                protected function deleteIndex()
                {
                    // TODO: Implement deleteIndex() method.
                }
            };
        });
    }
```

this leads to this result in the `/cp/utilities/search`

![image](https://user-images.githubusercontent.com/1016798/72226442-9126f180-3591-11ea-889a-08e8e9107a27.png)

I'd suggest to add the parameter to `callCustomCreator`, too.
This is a possible BC-Break if users already use the `$manager->extend(...)` functionality.

**Comment by jasonvarga** [comment]
Are you trying to create an index or a driver?

**Comment by psren** [comment]
An index. Similar to `Statamic\Search\Algolia\Index`.

**Comment by jasonvarga** [comment]
Are you trying to use another service?

**Comment by jasonvarga** [comment]
Nevermind, I see the issue. 👍

**Comment by geekish** [comment]
I've run into this while working on a Meilisearch driver. :(

**Comment by jasonvarga** [comment]
Addressed by #2781 - Sorry for the delay!

**Comment by HellPat** [comment]
@jasonvarga don't worry the delay and thank you very much!
